browser/src/Plugins/Plugin.ts

```diff
@@ -51,4 +50,31 @@ export class Plugin {
             Log.error(ex)
         }
     }
+
+    public hasTest(): boolean {
+        return (this._oniPluginMetadata && !!this._oniPluginMetadata.test)
+    }
+
+    public test(): void {
```

Comment:
This is a really creative idea! I wonder how it would scale to modules / plugins created outside the repo, though? For example - if `oni-plugin-markdown-preview` lived in a separate repo, we'd still want to test it. This also has a dependency on our settings for plugin paths, which might make it easy to miss a plugin under test. We'd probably want to be able to run `npm run test` on it, at least in that case - so I'm wondering if that would be sufficient for now.

Reply:
I think it should work/scale/evolve the same as the `activate` mechanism does.